airbyte-integrations/connectors/source-salesforce/source_salesforce/source.py

```diff
@@ -36,6 +37,26 @@ def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) ->
 
         return True, None
 
+    @classmethod
+    def _get_api_type(cls, stream_name, properties, stream_state):
+        # Salesforce BULK API currently does not support loading fields with data type base64 and compound data
+        properties_not_supported_by_bulk = {
+            key: value for key, value in properties.items() if value.get("format") == "base64" or "object" in value["type"]
+        }
+        properties_length = len(",".join(p for p in properties))
+
+        rest_required = stream_name in UNSUPPORTED_BULK_API_SALESFORCE_OBJECTS or properties_not_supported_by_bulk
+        # If we have a lot of properties we can overcome REST API URL length and get an error: "reason: URI Too Long".
+        # For such cases connector tries to use BULK API because it uses POST request and passes properties in the request body.
+        bulk_required = properties_length + 2000 > Salesforce.REQUEST_SIZE_LIMITS
+
+        if bulk_required and not rest_required:
```

Comment:
yes it possible!

my PR only improve current situation:
 - before: we used REST API because it's more prefereble for incremental sync 
 - after: we use BULK API because this concert stream has a large `properties_length` value

but yes, IT still possible we will catch potential problem in future

Reply:
for now we ignore such stream:
https://github.com/airbytehq/airbyte/blob/3faec782dac2b25097b09c1d493a15ab7ab07fc5/airbyte-integrations/connectors/source-salesforce/source_salesforce/source.py#L84

Reply:
I think it might make sense to just raise an exception in this case, actually - otherwise the user will be confused about why they got no data for these streams

Reply:
I have changed warning -> Exception